CardsCashCasino/Data/HoldEmUtil.cs

```diff
@@ -0,0 +1,335 @@
+
+            return new Tuple<List<Card>, Ranking>(bestHands[bestI], best);
+        }
+
+        /// <summary>
+        /// Used for breaking a tie between two hands of equal rank.
+        /// Assigns a value to hand1 that will be greater than some hand2, if hand1 has some card C such that 
+        ///     For each card in hand2 with a greater value, hand1 has another card with equal value.  
+        /// For example, the hand [A,2,2,2,2] Is worth a greater value than [K,K,K,K,Q]
+        /// And the hand [A,A,A,K,5] is worth a greater value than [A,A,K,K,K]
+        /// Hands [5,5,4,3,2] and [5,5,4,3,2] are considered tied and should result in a split pot.
+        /// </summary>
+        /// <param name="hand">A 5-card hand that was returned by GetScore.</param>
+        /// <returns>the value that will be compared to other values returned by this function to determine a winner.</returns>
+        public static long KickerValue(List<Card> hand)
+        {
+            Debug.Assert(hand.Count == 5);
+
+            // Special case: 2345A
+            if (hand[0].GetPokerValue() == 2 && hand[1].GetPokerValue() == 3 && hand[2].GetPokerValue() == 4 && hand[3].GetPokerValue() == 5 && hand[4].IsAce())
+            {
+                return (long)(Math.Pow(5, 5) + Math.Pow(4, 4) + Math.Pow(3, 3) + Math.Pow(2, 2) + Math.Pow(1, 1));
+            }
+
+            /*
+             * Decision for which hand wins in a tie is certainly overtuned but should be complete, 
+             *  and operates as if only Ranking.HIGH_CARD is possible.
+             * 
+             * Algorithm is simply :=   Score = 5^hand[4] + 4^hand[3] + ... + 1^hand[0];
+             * 12 12 12 12 12 =>    261,453,379
+             * 13 1  1  1  1  =>    1,220,703,135
+             * Which is what we want.
+             * 
+             * AAAAK    => 6,373,561,789
+             * INT_MAX  => 2,147,483,647
+             * So we will use 64 bit integer.
+             */
+            long handVal = 0;
+            for (int j = 4; j >= 0; j++)
+            {
+                handVal += (long)Math.Pow((j + 1), hand[j].GetPokerValue());
+            }
+            return handVal;
+        }
+
+
+        /// <summary>
+        /// Function to determine if a poker hand is a royal flush.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a royal flush, false otherwise.</returns>
+        private static bool IsRoyalFlush(List<Card> cards)
+        {
+            return IsStraightFlush(cards) && cards[0].GetPokerValue() == 10;
+        }
+
+        /// <summary>
+        /// Function to determine if a poker hand is a flush.
+        /// </summary>
+        /// <param name="cards">a 5-card hand.</param>
+        /// <returns>True if cards is a flush, false otherwise.</returns>
+        private static bool IsFlush(List<Card> cards)
+        {
+            Suit suit = cards[0].Suit;
+            foreach (Card card in cards)
```

Comment:
This can be done using Linq methods!

`return cards.All(card => card.Suit == cards[0].Suit).Count == 5;`